src/azure-cli-core/azure/cli/core/commands/arm.py

```diff
@@ -922,7 +922,7 @@ def remove_properties(instance, argument_values):
         list_to_remove_from = _find_property(instance, list_attribute_path)
         try:
             list_to_remove_from.pop(int(list_index))
-        except IndexError:
+        except (IndexError, AttributeError):
```

Comment:
If it is an AttributeError, do we need to compose a new error message?

Reply:
The error message is `index {} doesn't exist on {}`, I think it is also valid if the list is none.

Reply:
How about when update the property? There are multiple `IndexError` in this file, maybe double check all of them?

Reply:
> How about when update the property? There are multiple IndexError in this file, maybe double check all of them?

Confirmed 2 other places, both already catch AttributeError.

Reply:
I agree with @qwordy. Something like "_{list_attribute_path} doesn't exist_" sounds more reasonable.

Reply:
> I agree with @qwordy. Something like "{list_attribute_path} doesn't exist" sounds more reasonable.

message updated as recommend.